src/AgentVisibilityContext.tsx

```diff
@@ -30,71 +26,48 @@ export const VisibilityContext = createContext<VisibilityContextType>({
   highlightedAgents: {},
   receiveUIDisplayData: () => {},
   setHiddenAgents: () => {},
-  handleVisibilityCheckboxChange: () => {},
-  handleHightlightCheckboxChange: () => {},
+  handleHideCheckboxChange: () => {},
+  handleHideChildCheckboxChange: () => {},
+  handleHighlightChange: () => {},
+  handleChildHighlightChange: () => {},
 });
 
 export const VisibilityProvider = ({ children }: { children: ReactNode }) => {
   const [uiDisplayData, setUiDisplayData] = useState<UIDisplayData>([]);
-  const [hiddenAgents, setHiddenAgents] = useState<VisibilitySelectionMap>({});
-  const [highlightedAgents, setHighlightedAgents] =
-    useState<VisibilitySelectionMap>({});
+  const [hiddenAgents, setHiddenAgents] = useState<UserChangesMap>({});
+  const [highlightedAgents, setHighlightedAgents] = useState<UserChangesMap>(
+    {}
+  );
 
   const receiveUIDisplayData = (data: UIDisplayData) => {
     setUiDisplayData(data);
-    const noAgentsSelectedMap = mapUIDisplayDataToSelectionMap(data);
-    setHighlightedAgents(noAgentsSelectedMap);
-    setHiddenAgents(noAgentsSelectedMap);
+    const initialSelections = getInitialUserSelections(data);
+    setHighlightedAgents(initialSelections);
+    setHiddenAgents(initialSelections);
```

Comment:
```suggestion
    setHighlightedAgents(getInitialUserSelections(data));
    setHiddenAgents(getInitialUserSelections(data));
```

Maybe worth calling `getInitialUserSelections` twice, once for each of these setters, so they aren't sharing the same object reference? (I think it's fine based on how you're setting values, but it makes it clear that these aren't supposed to be sharing a map reference.)

Reply:
Yeah all the updates are functionalized so it shouldn't be an issue, and my default is to store something if you're going to use the same value twice?

But maybe we can split the different on the two suggestions here and keep the function name and drop the shared variable. I'll push that change.